libvast/src/detail/pid_file.cpp

```diff
@@ -52,9 +66,9 @@ caf::error acquire_pid_file(const path& filename) {
                       "failed in flock(2):", strerror(errno));
   }
   // Write the PID in human readable form into the file.
-  auto pid = to_string(process_id());
-  VAST_ASSERT(!pid.empty());
-  if (::write(fd, pid.data(), pid.size()) < 0) {
+  auto pid_string = to_string(pid);
+  VAST_ASSERT(!pid_string.empty());
+  if (::write(fd, pid_string.data(), pid_string.size()) < 0) {
```

Comment:
We could use `io::write` here, now that we are touching this piece of code.

Reply:
We won't be able to use `flock` if we do that.

Reply:
Ah, right.